### PR TITLE
Connect litter pools to soil pools

### PR DESCRIPTION
**Pull request by jacobcook1995**
# Description

This pull request connects the litter pools to the soil pools, so that when litter decays a fraction of it enters the soil. I've also added to the example data so that the litter model can be run as part of `vr_run`.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Key checklist

- [x] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [x] All tests pass: `$ poetry run pytest`

## Further checks

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: virtual_rainforest/models/litter/constants.py [hunk 1]
```diff
@@ -54,3 +56,23 @@ class LitterConsts:
     This value was estimated (very unsystematically) from SAFE project data. This
     constant will eventually be removed once the litter is linked to other models.
     """
+
+    cue_metabolic: float = 0.45
+    """Carbon use efficiency of metabolic litter decay [unitless].
+
+    The value given here is taken from :cite:t:`fatichi_mechanistic_2019`, but I can't
+    track down an empirical source.This carbon use efficiency is constant with
```
Comment (alexdewar) [inline virtual_rainforest/models/litter/constants.py:64]:
```suggestion
    track down an empirical source. This carbon use efficiency is constant with
```

File: virtual_rainforest/models/litter/constants.py [hunk 2]
```diff
@@ -54,3 +56,23 @@ class LitterConsts:
...
+    """
+
+    cue_structural_above_ground: float = 0.55
+    """Carbon use efficiency of aboveground structural litter decay [unitless].
+
+    The value given here is taken from :cite:t:`fatichi_mechanistic_2019`, but I can't
+    track down an empirical source. This carbon use efficiency is constant with
+    temperature, soil moisture and substrate stoichiometry. These assumptions are not
+    made in the soil model, but are used for the sake of simplicity here. If an improved
+    version of the litter model gets made, this is a key area to address.
```
Comment (alexdewar) [inline virtual_rainforest/models/litter/constants.py:77]: I wouldn't just copy and paste this text for the docstrings.

```suggestion
    The value given here is taken from :cite:t:`fatichi_mechanistic_2019`; see documentation for `cue_metabolic` 
    for details.
```

**Review by alexdewar — APPROVED** [review]
I've given this a quick once-over and it seems sensible to me. I'm a little lost on the science here though, so I'm not really able to give a more in-depth review.

**Comment by jacobcook1995** [comment]
Ahh cheers @alexdewar, I've been adding you and Diego to fairly sciencey pull requests mainly because I want to check that bad code style isn't creeping in, so a once over to check that there isn't anything glaringly bad is very helpful!

File: docs/source/data_recipes/litter_dummy.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  name: vr_python3
+---
+
+# Necessary litter data for dummy module
+
+This section explains how to generate the data required to run the litter component of
+the dummy model. **It is important to note that none of this data is real data**.
+Instead, this data is a set of plausible values for which the litter model absolutely
+has to function sensibly for.
+
```
Comment (vgro) [inline docs/source/data_recipes/litter_dummy.md:23]: could you point to the page where you describe the different litter pools?

File: virtual_rainforest/models/soil/carbon.py [hunk 4]
```diff
@@ -40,14 +41,15 @@ def calculate_soil_carbon_updates(
         soil_moisture: relative water content for each soil grid cell [unitless]
         soil_temp: soil temperature for each soil grid cell [degrees C]
         percent_clay: Percentage clay for each soil grid cell
+        mineralisation_rate: Amount of litter mineralised into POM pool [kg C m^-3
+            day^-1]
```
Comment (vgro) [inline virtual_rainforest/models/soil/carbon.py:45]: should this be m^2?

**Review by vgro — APPROVED** [review]
LGTM! I'm also not very familiar with the soil carbon science, but the general approach looks sensible :-)

**Pull request merged**